src/platform/plugins/shared/workflows_execution_engine/server/step/step-base.ts

```diff
@@ -0,0 +1,67 @@
+import { WorkflowContextManager } from '../workflow-context-manager/workflow-context-manager';
+import { BaseStepSchema } from '@kbn/workflows';
+// Import specific step types as needed from schema
+import { z } from '@kbn/zod';
+// import { evaluate } from '@marcbachmann/cel-js'
+import { WorkflowTemplatingEngine } from '../templating-engine';
+import { ConnectorExecutor } from '../connector-executor';
+
+export interface RunStepResult {
+    output: any;
+    error: any;
+}
+
+export type StepDefinition = z.infer<typeof BaseStepSchema>; // Base type; subclasses will use specific inferred types
+
+export abstract class StepBase {
+    protected step: StepDefinition;
+    protected contextManager: WorkflowContextManager;
+    protected templatingEngine: WorkflowTemplatingEngine;
+    protected connectorExecutor: ConnectorExecutor;
+
+    constructor(
+        step: StepDefinition,
+        contextManager: WorkflowContextManager,
+        connectorExecutor: ConnectorExecutor,
+        templatingEngineType: 'mustache' | 'nunjucks' = 'nunjucks',
+    ) {
+        this.step = step;
+        this.contextManager = contextManager;
+        this.templatingEngine = new WorkflowTemplatingEngine(templatingEngineType);
+        this.connectorExecutor = connectorExecutor;
+    }
+
+    public getName(): string {
+        return this.step.name;
+    }
+
+    public async run(): Promise<RunStepResult> {
```

Comment:
Wondering if we need any result to be returned from "run" function given that we have context manager that will be  populated from step runners.